classes/helper/mahara_portfolio.php

```diff
@@ -0,0 +1,273 @@
+                'component' => 'plagiarism_urkund',
+                'filearea' => 'htmllite',
+                'itemid' => $this->submission->id,
+                'filepath' => '/maharahtmllite/',
+                'filename' => 's' . $this->submission->id . '.zip'
+        );
+        $this->fileinfo = $fileinfo;
+
+        if ($file = $fs->get_file($fileinfo['contextid'], $fileinfo['component'], $fileinfo['filearea'],
+                $fileinfo['itemid'], $fileinfo['filepath'], $fileinfo['filename'])) {
+            mtrace('urkund: mahara_portfolio: Using previously downloaded zip file.');
+            $this->zipfile = $file;
+        }
+    }
+
+    /**
+     * Get Mahara submission information from the database
+     *
+     * @param  int $submissionid
+     * @return mixed
+     */
+    private function get_mahara_submission($submissionid) {
+        global $DB;
+
+        return $DB->get_record('assignsubmission_maharaws', array('submission' => $submissionid));
+    }
+
+    /**
+     * Run all the steps to download a portfolio and add it to the Ouriginal queue.
+     *
+     */
+    public function download_to_queue() {
+        if (empty($this->zipfile)) {
+            $this->get_fileurl();
+            $this->download_zip();
+        }
+        $this->queue_files();
+    }
+
+    /**
+     * Make initial Mahara HTML Lite API request to generate files for the portfolio.
+     *
+     */
+    private function get_fileurl() {
+        mtrace('urkund: mahara_portfolio: Calling Mahara HTML Lite API.');
+
+        // Get fileurl.
+        $params = array('viewid' => $this->submission->viewid,
+                'iscollection' => $this->submission->iscollection,
+                'submittedhost' => $this->host,
+                'exporttype' => 'htmllite');
+        $params = array('views' => array($params));
+        $data = $this->webservice_call('mahara_submission_generate_view_for_plagiarism_test', $params, $method = "POST");
+        $this->fileurl = $data[0]['fileurl'];
+    }
+
+    /**
+     * Webservice call helper.
+     *
+     * @param string $function
+     * @param array $params
+     * @param string $method
+     *
+     * @return mixed
+     */
+    private function webservice_call($function, $params, $method = "POST") {
+        global $CFG;
+
+        $endpoint = $this->oauthurl .
+            (preg_match('/\/$/', $this->oauthurl) ? '' : '/') .
+            'webservice/rest/server.php';
+        $args = array(
+            'oauth_consumer_key' => $this->oauthkey,
+            'oauth_consumer_secret' => $this->oauthsecret,
+            'oauth_callback' => 'about:blank',
+            'api_root' => $endpoint,
+        );
+
+        // Reuse the maharaws oauth client.
+        require_once($CFG->dirroot . '/mod/assign/submissionplugin.php');
+        require_once($CFG->dirroot . '/mod/assign/submission/maharaws/locallib.php');
+        $client = new \mahara_oauth($args);
+        if (!empty($CFG->disablesslchecks)) {
```

Comment:
what's this for? - typically I have a hate for this sort of workaround... especailly if it's for some internal testing site... just use proper SSL certs - don't let people avoid them.